src/olympia/access/acl.py

```diff
@@ -149,8 +149,13 @@ def check_addon_ownership(request, addon, dev=False, admin=True, ignore_disabled
     # Users with 'Addons:Edit' can do anything.
     if admin and action_allowed(request, amo.permissions.ADDONS_EDIT):
         return True
-    # Only admins can edit admin-disabled addons.
-    if addon.status == amo.STATUS_DISABLED and not ignore_disabled:
+    # Only admins can edit admin-disabled addons or permission-enabler add-ons,
+    # unless the caller determined it was ok (for specific scenarios like
+    # read-only requests).
+    if not ignore_disabled and (
```

Comment:
I was afraid you'd say that :) I didn't want to touch the signature because `ignore_disabled` is also a parameter in the more generic `check_ownership()` function (and methods on some objects like `Collection`), but yeah, I need to do something about that cause it's all terrible...

Reply:
https://github.com/mozilla/addons-server/issues/18444 is the follow-up